plugins/woocommerce-blocks/assets/js/data/cart/persistence-layer.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import type { Cart } from '@woocommerce/types';
+
+function getCookie( name: string ): string | Record< string, string > {
+	const cookies = document.cookie
+		.split( ';' )
+		.reduce< Record< string, string > >( ( acc, cookieString ) => {
+			const [ key, value ] = cookieString
+				.split( '=' )
+				.map( ( s ) => s.trim() );
+			if ( key && value ) {
+				acc[ key ] = decodeURIComponent( value );
+			}
+			return acc;
+		}, {} );
+	return name ? cookies[ name ] || '' : cookies;
+}
+
+export const hasCartSession = () => {
+	return !! getCookie( 'woocommerce_items_in_cart' );
+};
+
+export const isAddingToCart = () => {
+	return !! window.location.search.match( /add-to-cart/ );
+};
+
+export const persistenceLayer = {
```

Comment:
This handles getting and setting data. The core persistence plugin did not work for some reason, but we also have some edge cases (adding to cart/no cookie) where the cache is not needed.